### PR TITLE
Patch 2

### PR DESCRIPTION
**Pull request by savjani**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 101-managed-mysql-with-vnet/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,199 @@
+{
```
Comment (bmoore-msft) [inline 101-managed-mysql-with-vnet/azuredeploy.json]: format the JSON, e.g. SHIFT+ALT+F in VS Code

File: 101-managed-mysql-with-vnet/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        "description": "MySQL Server backup retention days"
+     }
+     },
+    "geoRedundantBackup": {
+    "type": "string",
+    "defaultValue": "Disabled",
+    "metadata": {
+        "description": "Geo-Redundant Backup setting"        }            
+    },
+    "virtualNetworkName": {
```
Comment (bmoore-msft) [inline 101-managed-mysql-with-vnet/azuredeploy.json]: add a defaultValue

File: 101-managed-mysql-with-vnet/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            }
+        },
+        "sku": {
+            "name": "[parameters('skuName')]",
+            "tier": "[parameters('skuTier')]",
+            "capacity": "[parameters('skuCapacity')]",
+            "size": "[parameters('skuSizeMB')]",
+            "family": "[parameters('skuFamily')]"
+        },
+        "type": "Microsoft.DBforMySQL/servers"
```
Comment (bmoore-msft) [inline 101-managed-mysql-with-vnet/azuredeploy.json]: move this to the top of the object

File: 101-managed-mysql-with-vnet/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+            "version": "[parameters('mysqlVersion')]",
+            "administratorLogin": "[parameters('administratorLogin')]",
+            "administratorLoginPassword": "[parameters('administratorLoginPassword')]",
+            "storageProfile": {
+                "storageMB": "[parameters('skuSizeMB')]",
+                "backupRetentionDays": "[parameters('backupRetentionDays')]",
+                "geoRedundantBackup": "[parameters('geoRedundantBackup')]"
+            }
+        },
+        "sku": {
```
Comment (bmoore-msft) [inline 101-managed-mysql-with-vnet/azuredeploy.json]: move this above properties

File: 101-managed-mysql-with-vnet/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+            "properties": {
+              "addressPrefix": "[parameters('subnetPrefix')]"
+            }
+        }
+    ]
+}, 
+    
+    {
+        "apiVersion": "2017-12-01",
+        "kind": "",
```
Comment (bmoore-msft) [inline 101-managed-mysql-with-vnet/azuredeploy.json]: remove this

File: 101-managed-mysql-with-vnet/azuredeploy.parameters.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {   
+    "serverName": {
+      "value": "GEN-UNIQUE"
+    }, 
+    "administratorLogin": {
+      "value": "mysqladmin"
```
Comment (bmoore-msft) [inline 101-managed-mysql-with-vnet/azuredeploy.parameters.json]: GEN-UNIQUE

Comment (savjani) [reply 101-managed-mysql-with-vnet/azuredeploy.parameters.json]: GEN-UNIQUE was generating values greater than 16 length which caused the build to fail. I have reverted to mysqladmin. Let me know if you have other suggestions

**Comment by savjani** [comment]
@bmoore-msft - Any update on this PR?

**Pull request merged**